packages/query/src/query/useList.ts

```diff
@@ -147,7 +147,7 @@ const withSort = () => (props: UseListProps & UseListFilters) => {
   const { order } = props
   const [sortKey, sortDirection] = getSort(props)
 
-  if (order || !sortKey) return props
+  if (!(order && sortKey)) return props
```

Comment:
should this be `if (order || !sortKey) return props`? if `order` is specified OR if sortKey is not specified, we dont construct a new order and return the original props (since either `order` is provided, or cannot construct a new order since`sortKey` is not provided).

Reply:
`!(order && sortKey)` == `!order || !sortKey`, so we would return the original props if `order` is not provided, but `sortKey` might be provided -> user ends up with no `order` in the returned props

Reply:
tested it out and I agree with this, found that the problem is elsewhere. thanks hieu!